gflow/gflow.py

```diff
@@ -1,107 +1,255 @@
+        Returns:
+            results (LibraryDataset): Dataset object that represents the uploaded content if successful,
+                                      None if not successful
+        """
+        results = None
+        self.logger.debug("Datasets source: '%s'" % self.datasets_source)
         for i in range(0, len(self.datasets)):
-            self.logger.debug("Dataset source: '%s'" % self.datasets[i].input_type)
-            if self.datasets[i].input_type == 'local':
-                library.upload_from_local(self.datasets[i].name)
-            elif self.datasets[i].input_type == 'url':
-                library.upload_from_url(self.datasets[i].name)      # Need a URL to test
-            else:
+            self.logger.debug("Uploading dataset: " + self.datasets[i])
+            if self.datasets_source == 'local':
                 try:
-                    library.upload_from_galaxy_fs(self.datasets[i].name)        # Still doesn't work
-                except:
-                    self.logger.error("File upload unsuccessful, only admins can "
-                                      "upload files from the Galaxy filesystem")
-                    e = sys.exc_info()[0]
+                    results = library.upload_from_local(self.datasets[i])
+                except IOError as e:
                     self.logger.error(e)
-                    sys.exit(1)
-        return 0
-
-    def set_tool_params(self):
+                    raise IOError
+            elif self.datasets_source == 'url':
+                results = library.upload_from_url(self.datasets[i])  # Need a URL to test
+            else:
+                self.logger.error("Dataset source must be local or url")
+                raise ValueError
+        return results
+
+    def set_runtime_params(self, wf):
+        """
+        Map the parameters of tools requiring runtime parameters to the step ID of each tool.
+
+        Args:
+            wf (Workflow): The workflow object containing the tools
+        Returns:
+            params (dict): The dictionary containing the step IDs and parameters.
+        """
         params = {}
-        for i in range(0, self.options.num_tools):
-            params = {self.options.step_ids[i]: self.options.tool_params[i]}
+        for i in range(0, len(self.runtime_params)):
+            param_dict = {}
+            for j in range(0, len(self.runtime_params['tool_' + str(i)])):
+                param_dict[self.runtime_params['tool_' + str(i)]['param_' + str(j)]['name']] \
+                    = self.runtime_params['tool_' + str(i)]['param_' + str(j)]['value']
+                for s in wf.sorted_step_ids():
+                    try:
+                        if wf.steps[s].tool_inputs[self.runtime_params['tool_' + str(i)]['param_' + str(j)]['name']]:
+                            params[s] = param_dict
+                    except KeyError:
+                        pass
         return params
 
     def run_workflow(self):
```

Comment:
Since your class is "galaxy work flow", then this method can just be run(), no need to add "workflow"